### PR TITLE
feat: show all available organizations when creating project grants 

### LINKED ISSUE #5317: Show all available organizations when creating (project) grants
As a authorized Manager, I want see all available (to me) organizations so that I can create a project grant.

Considerations
- At the moment I have to go to the org overview on instance level and copy the domain string to enter in the field
- Search for "Customer" does not give me the expected result (I have two orgs called Demo-Customer and Demo-Customer2)

Screenshot 2023-02-28 at 13 33 06

```[tasklist]
### Acceptance Criteria
- [ ] All available organizations are shown (based on user authorizations) directly on the first screen
- [ ] Search can filter for partial matches
```

### PR DESCRIPTION
**Pull request by doncicuto**
As an authorized Manager, I want see all available (to me) organizations so that I can create a project grant.

This is the proposed scenario where is no longer needed to enter the domain of the organization you want to grant (go to the organization detail page when you can't remember it):

https://github.com/zitadel/zitadel/assets/30386061/a5cee5cf-d5dc-4c28-b30c-663966d20362

Acceptance Criteria

- [X] All available organizations are shown (based on user authorizations) directly on the first screen
- [X] Search can filter for partial matches

Closes #5317 

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [X] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [ ] Critical parts are tested automatically
- [X] Where possible E2E tests are implemented
- [X] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by hifabienne** [comment]
@peintnermax can you please have a look at this?

File: console/src/app/modules/search-org-autocomplete/search-org-autocomplete.component.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      });
+  }
+
+  public ngOnInit(): void {
+    const query = new OrgQuery();
+    const orgStateQuery = new OrgStateQuery();
+    orgStateQuery.setState(OrgState.ORG_STATE_ACTIVE);
+    query.setStateQuery(orgStateQuery);
+
+    this.auth.listMyProjectOrgs(undefined, 0, query ? [query] : undefined).then((orgs) => {
```
Comment (peintnermax) [inline console/src/app/modules/search-org-autocomplete/search-org-autocomplete.component.ts:63]:
```suggestion
    this.auth.listMyProjectOrgs(undefined, 0, [query]).then((orgs) => {
```

lets clean this up a little

File: console/src/app/modules/search-org-autocomplete/search-org-autocomplete.component.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          query.setStateQuery(orgStateQuery);
+
+          if (value) {
+            const orgNameQuery = new OrgNameQuery();
+            orgNameQuery.setName(value);
+            orgNameQuery.setMethod(TextQueryMethod.TEXT_QUERY_METHOD_CONTAINS_IGNORE_CASE);
+            query.setNameQuery(orgNameQuery);
+          }
+
+          return from(this.auth.listMyProjectOrgs(undefined, 0, query ? [query] : undefined)).pipe(
```
Comment (peintnermax) [inline console/src/app/modules/search-org-autocomplete/search-org-autocomplete.component.ts:44]:
```suggestion
          const stateQuery = new OrgQuery();

          const orgStateQuery = new OrgStateQuery();
          orgStateQuery.setState(OrgState.ORG_STATE_ACTIVE);
          stateQuery.setStateQuery(orgStateQuery);

          let queries: OrgQuery[] = [stateQuery];

          if (value) {
            const nameQuery = new OrgQuery();

            const orgNameQuery = new OrgNameQuery();
            orgNameQuery.setName(value);
            orgNameQuery.setMethod(TextQueryMethod.TEXT_QUERY_METHOD_CONTAINS_IGNORE_CASE);

            nameQuery.setNameQuery(orgNameQuery);
            queries = [stateQuery, nameQuery];
          }

          return from(this.auth.listMyProjectOrgs(undefined, 0, queries)).pipe(
```
`OrgQuery` is a one of so we need to create a seperate query for the name

**Review by peintnermax — CHANGES_REQUESTED** [review]
As the `OrgQuery` is a one of, we need to provide an array.. The rest is amazing work. Thx 👍

**Comment by doncicuto** [comment]
Thanks @peintnermax for your time reviewing my code. I've uploaded your requested changes

**Comment by doncicuto** [comment]
Hi @peintnermax as GitHub still shows that there are pending requested changes, I'm not sure if you need me to do more changes to this PR. If that is the case, please let me know. Thanks as always

**Review by peintnermax — APPROVED** [review]

**Pull request merged**